### ISSUE TITLE
Application breaks on IE11 with Ember.js 3.6

### ISSUE THREAD

**Issue by YoranBrondsema** [emberjs/ember.js#17497]
Yesterday we upgraded our app to Ember.js 3.6 but noticed that that broke it on IE11. We've had IE11 set as a target in `config/targets.js` this whole time.

It breaks on the first `define` in `app.js`, i.e. the bundle containing all the application-specific Javascript. The error is:

`'define' is undefined`

I see that PRs https://github.com/emberjs/ember.js/pull/17188 and https://github.com/emberjs/ember.js/pull/17246 might have something to do with it. I checked the value of `needsLegacyBuild` from https://github.com/emberjs/ember.js/blob/v3.6.1/lib/index.js#L84 during the build and it's set to `true`, as you would expect. So I'm not too sure about where the bug is.

**Comment by YoranBrondsema** [comment]
That's the first error I saw, I think. I reverted back to 3.5 now so can't check right away, but I'll work on a reproduction app tomorrow and get back to you.

**Comment by rwjblue** [comment]
Awesome, thank you!

**Comment by YoranBrondsema** [comment]
OK I figured it out. In the end it was my own fault, because I imported the template compiler with

```
app.import('node_modules/ember-source/dist/ember-template-compiler.js');
```

instead of

```
app.import('vendor/ember/ember-template-compiler.js');
```

How come it didn't do the transpilation when importing directly from `node_modules`? I'm not too familiar with the internals.

I figured it out through a StackOverflow answer. Since the template compiler is part of the core, maybe this is something that belongs in the guide? For instance on https://guides.emberjs.com/release/addons-and-dependencies/managing-dependencies/? I would be happy to open up a PR to document the proper way to import the template compiler somewhere in the docs.
